vllm/v1/core/single_type_kv_cache_manager.py

```diff
@@ -532,29 +532,149 @@ def find_longest_cache_hit(
         assert isinstance(
             kv_cache_spec,
             MambaSpec), ("MambaManager can only be used for mamba groups")
-        # Prefix caching is not supported for mamba now. Always return empty
-        # list.
         computed_blocks: tuple[list[KVCacheBlock], ...] = tuple(
             [] for _ in range(len(kv_cache_group_ids)))
+        if kv_cache_spec.cache_strategy == "disabled":
+            return computed_blocks #return empty list if cache is disabled
+
+        max_num_blocks = max_length // kv_cache_spec.block_size
+        # Search from right to left and early stop when a match is found.
+        for i in range(max_num_blocks - 1, -1, -1):
+            if cached_block := block_pool.get_cached_block(
+                    block_hashes[i], kv_cache_group_ids):
+                for computed, cached in zip(computed_blocks, cached_block):
+                    # the hit length logic later assumes: 
+                    #  hit_length = len(hit_blocks_other_attn[0]) 
+                    #               * self.other_block_size
+                    # so we insert dummy blocks at the beginning:
+                    if i > 0:
+                        computed.extend([block_pool.null_block] * i)
+                    computed.append(cached)
+                break # we just need the last match - early stopping
+                
         return computed_blocks
 
     def remove_skipped_blocks(self, request_id: str,
                               num_computed_tokens: int) -> None:
-        # Each request will always have 1 block at this moment, so no need to
-        # remove blocks.
+        # TODO: For "all" strategy, and potentially for "last"
+        #  we should already start removing initial blocks
```

Comment:
Why would we want to remove initial blocks? I thought we would want to just keep everything around and then if things fill up we evict the least recently used ones.

Reply:
We can't evict a block that is using by running requests. This function is for freeing the initial blocks in a running request so that these blocks are evitable and can be allocated to other requests.